pkgs/development/python-modules/qtile-bonsai/default.nix

```diff
@@ -0,0 +1,68 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  cairocffi,
+  cffi,
+  strenum,
+  psutil,
+  xcffib,
+  pdm-backend,
+  pyside6,
+  pyvirtualdisplay,
+  qtile,
+}:
+
+buildPythonPackage rec {
+  pname = "qtile-bonsai";
+  version = "0.4.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "aravinda0";
+    repo = "qtile-bonsai";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-IWy/YEVdZc+UgIKl75ZpOkOIvpS5hCX0ihQenUOuJHo=";
+  };
+
+  build-system = [
+    pdm-backend
+  ];
+
+  dependencies = [
+    strenum
+    psutil
+  ];
+
+  nativeCheckInputs = [
+    pyside6
+    pyvirtualdisplay
+    (cairocffi.override { withXcffib = true; })
+    cffi
+    xcffib
+    qtile
+  ];
```

Comment:
Please add `pytestCheckHook` to nativeCheckinputs or provide `checkPhase` to enable tests.
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pytestcheckhook-using-pytestcheckhook

Reply:
done